opm/simulators/wells/WellInterfaceFluidSystem.cpp

```diff
@@ -836,10 +836,21 @@ checkMaxRatioLimitCompletions(const SingleWellState& ws,
         }
     } // end of for (const auto& completion : completions_)
 
-    assert(max_ratio_completion > max_ratio_limit);
-    assert(worst_offending_completion != INVALIDCOMPLETION);
+    // No worst offending completion is found because all the completions are either injecting or
+    // have trivial rates.
+    if (worst_offending_completion == INVALIDCOMPLETION) {
+        report.ratio_limit_violated = false;
+        return;
+    }
+
+    // Due to numerical instability there may exist corner cases where the well breaks
+    // the ratio limit but no completion does for wells with very small rates.
+    if (max_ratio_completion <= max_ratio_limit) {
```

Comment:
Thanks for addressing the comments.  

This line is also a special situation we do not see normally, that noisy small values make the general assumption does not apply anymore. It is a little different from the situation that with the only completion is injecting. 

And changing the value `report.ratio_limit_violated` at the this stage is pretty dramatic. It can be good to give some message `debug` or even` warning` to let people know that something extraordinary going on at these two occasions. 

I have no more further comments.

Reply:
I have added a warning as you suggested. We may want to move it to debug if this happens more often than we think. Thanks for the review.